File: final_asssignment_backend_go/project/internal/domain/fine_information.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// FineInformation 表示 fine_information 表的实体
type FineInformation struct {
	FineID        int            `gorm:"column:fine_id;primaryKey;autoIncrement" json:"fine_id"`
	OffenseID     int            `gorm:"column:offense_id" json:"offense_id"`
	FineAmount    float64        `gorm:"column:fine_amount" json:"fine_amount"`
	FineTime      time.Time      `gorm:"column:fine_time" json:"fine_time"`
	Payee         string         `gorm:"column:payee" json:"payee"`
	AccountNumber string         `gorm:"column:account_number" json:"account_number"`
	Bank          string         `gorm:"column:bank" json:"bank"`
	ReceiptNumber string         `gorm:"column:receipt_number" json:"receipt_number"`
	Remarks       string         `gorm:"column:remarks" json:"remarks"`
	DeletedAt     gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (FineInformation) TableName() string {
	return "fine_information"
}

File: final_assignment_backend_quarkus/src/main/java/finalassignmentbackend/entity/FineInformation.java
package finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 罚款信息类
 * 该类用于表示数据库中"fine_information"表的记录，包含罚款的相关信息
 */
@Data
@TableName("fine_information")
public class FineInformation implements Serializable {
    // 用于序列化和反序列化时的版本标识
    @Serial
    private static final long serialVersionUID = 1L;

    @TableId(value = "fine_id", type = IdType.AUTO)
    private Integer fineId;

    @TableField("offense_id")
    private Integer offenseId;

    // 罚款金额
    @TableField("fine_amount")
    private BigDecimal fineAmount;

    // 罚款时间
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField("fine_time")
    private LocalDateTime fineTime;

    // 缴费人姓名
    @TableField("payee")
    private String payee;

    // 银行账号
    @TableField("account_number")
    private String accountNumber;

    // 银行名称
    @TableField("bank")
    private String bank;

    // 收据编号
    @TableField("receipt_number")
    private String receiptNumber;

    // 备注信息
    @TableField("remarks")
    private String remarks;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
}